Guide rework (#298)
rstudio/rsconnect-jupyter: docs/docs/usage.md

to publish the current notebook to RStudio Connect.

!!! note
    This plugin is only for notebooks using Python kernels. Therefore, R notebooks cannot be published using this plugin.

## Entering server information

- If this is your first time publishing a notebook, you will be
prompted to enter the location and a nickname for the RStudio Connect server.
- You will also be prompted to enter your API Key. See the [RStudio Connect User
Guide](http://docs.rstudio.com/connect/user/api-keys) for
instructions on generating API Keys for your user.
- When you click the **Add Server** button, `rsconnect-jupyter` will send a request to the RStudio Connect server to verify that it can be reached via the requested URL and that the API key is valid.

If your RStudio Connect server was configured with a self-signed certificate (or other certificates that the computer hosting your Jupyter notebook server does not trust), the attempt to contact RStudio Connect may fail with a TLS-related error.

You have multiple options in this case, depending on your needs:

> 1. If your RStudio Connect Administrator can give you the Certificate Authority (CA)

Reviewer: Do the administrator references here need to be product-specific?

Reviewer 2: Change to "administrator"?

kernel environment; that is, the environment where the `ipykernel` package is
installed. In most cases that will be the same as the notebook server
environment where `jupyter` is installed.

The command `pip freeze` will be used to inspect the environment. The output
of `pip freeze` lists all packages currently installed, as well as their
versions, which enables RStudio Connect to recreate the same environment.

## Generating Manifests for git Publishing

RStudio Connect can poll git repositories for deployable content and update
as you add new commits to your repository. To be deployable, a
directory must have a valid `manifest.json`. Python content should also have
some kind of environment file (i.e.: `requirements.txt`) to be able
to restore the package set in your current environment.

To begin, click the **Publish** button and select **Create Manifest for git Publishing**.

<img class="border" src="../images/deploy-options.png" class="block" alt="Deployment drop-down
menu showing "Publish to Connect" and "Create Manifest for git Publishing>

When you click **Create Manifest**, one of the following will happen:

> - If a `manifest.json` and `requirements.txt` file does not exist, they will be generated for the current notebook using your current environment.

Reviewer: ```suggestion
- If a `manifest.json` or a `requirements.txt` file does not exist, one will be generated for the current notebook using your current environment.
```

> - If they do exist, you will be presented with a message
> informing you of this fact. If you need to regenerate the files, delete them in the Jupyter UI or using the console, then repeat this process.

Reviewer: ```suggestion
- If either file exists, you will be presented with a message
informing you of this fact. If you need to regenerate the files, delete them in the Jupyter UI or using the console, then repeat this process.
```
